pkg/domain/infra/abi/quadlet.go

```diff
@@ -0,0 +1,764 @@
+			if err != nil {
+				return nil, fmt.Errorf("unable to read Quadlet dir %s: %w", pathsOrURLs[0], err)
+			}
+			redoPaths := []string{}
+			for _, entry := range entries {
+				redoPaths = append(redoPaths, filepath.Join(pathsOrURLs[0], entry.Name()))
+			}
+			redoPaths = append(redoPaths, pathsOrURLs[1:]...)
+			paths = redoPaths
+			// treat all file in this session as part of one app.
+			assetFile = "." + filepath.Base(pathsOrURLs[0]) + ".app"
+		}
+	}
+
+	// Loop over all given URLs
+	for _, toInstall := range paths {
+		validateQuadletFile := false
+		if assetFile == "" {
+			assetFile = "." + filepath.Base(toInstall) + ".asset"
+			validateQuadletFile = true
+		}
+		switch {
+		case strings.HasPrefix(toInstall, "http://") || strings.HasPrefix(toInstall, "https://"):
+			r, err := http.Get(toInstall)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("unable to download URL %s: %w", toInstall, err)
+				continue
+			}
+			defer r.Body.Close()
+			quadletExtension := getFileExtension(r)
+			// It's a URL. Pull to temporary file.
+			tmpFile, err := os.CreateTemp("", "quadlet-dl-*"+quadletExtension)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("unable to create temporary file to download URL %s: %w", toInstall, err)
+				continue
+			}
+			defer func() {
+				tmpFile.Close()
+				if err := os.Remove(tmpFile.Name()); err != nil {
+					logrus.Errorf("Unable to remove temporary file %q: %v", tmpFile.Name(), err)
+				}
+			}()
+			_, err = io.Copy(tmpFile, r.Body)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = fmt.Errorf("populating temporary file: %w", err)
+				continue
+			}
+			installedPath, err := ic.installQuadlet(ctx, tmpFile.Name(), "", installDir, assetFile, validateQuadletFile)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = err
+				continue
+			}
+			installReport.InstalledQuadlets[toInstall] = installedPath
+		default:
+			err := fileutils.Exists(toInstall)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = err
+				continue
+			}
+			// If toInstall is a single file, execute the original logic
+			installedPath, err := ic.installQuadlet(ctx, toInstall, "", installDir, assetFile, validateQuadletFile)
+			if err != nil {
+				installReport.QuadletErrors[toInstall] = err
+			} else {
+				installReport.InstalledQuadlets[toInstall] = installedPath
+			}
+		}
+	}
+
+	// TODO: Should we still do this if the above validation errored?
+	if options.ReloadSystemd {
+		if err := conn.ReloadContext(ctx); err != nil {
+			return &installReport, fmt.Errorf("reloading systemd: %w", err)
+		}
+	}
+
+	return &installReport, nil
+}
+
+// Extracts file extension from Content-Disposition or URL
+func getFileExtension(resp *http.Response) string {
+	// Try Content-Disposition header first
+	cd := resp.Header.Get("Content-Disposition")
+	if cd != "" {
+		re := regexp.MustCompile(`(?i)filename="?([^"]+)"?`)
```

Comment:
see my previous comment about this, https://github.com/containers/podman/pull/26330#discussion_r2182934570 which has a well defined function on how to parse this header

Also then return the full filename so the caller can use this as final quadlet name at least.